boards/arm/stm32f746g_disco/Kconfig.defconfig

```diff
@@ -0,0 +1,28 @@
+# Kconfig - STM32F746G DISCOVERY board configuration
+#
+# Copyright (c) 2018 Yurii Hamann
+
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if BOARD_STM32F746G_DISCO
+
+config BOARD
+	default stm32f746g_disco
+
+if UART_CONSOLE
+
+config UART_STM32_PORT_1
+	default y
+
+endif # UART_CONSOLE
+
+if GPIO_STM32
```

Comment:
Move this to SoC (GPIO Commit)

Reply:
Why do we need to mix board and SoC code in commit?

Reply:
Good question.
In order to avoid confusing variety of GPIO configurations in various boards, it has been decided to enable all GPIO ports as default board configuration (documented here: http://docs.zephyrproject.org/porting/board_porting.html#default-board-configuration)
Then, since all available GPIO are enabled, this no longer depends on board, and can be moved to SoC.
Then, if user wants to deactivate some port as part of use case optimization (footprint and power), this should be done as application configuration.

Reply:
How should new board be added?
It's necessary to create one commit with board basic support and another commit that enables GPIO?

Reply:
Let's activate GPIOs as part of the GPIO commit.
GPIOs A, B and C are already enabled in arch/arm/soc/st_stm32/common/Kconfig.defconfig.series. Do nothing for these.
In arch/arm/soc/st_stm32/stm32f7/Kconfig.defconfig.series, enable ports D E F G H and I as they seems to be always available in all F7 SoCs
Then, in matching Kconfig.defconfig.stm32f7xxxx file, enable the additional GPIOs if any (J K)

Reply:
Now I have understood what you meant. Thank you.